### PR TITLE
#166240987 setup API documentation using swagger

### PR DESCRIPTION
**Pull request by EmyRukundo**
**What does this PR do?**
      
  swagger documentation

**Description of Task to be completed?**

   setup API documentation using swagger

**How should this be manually tested?**

N/A

 **Any background context you want to provide?**

N/A

**What are the relevant pivotal tracker stories?**

 [#166240987](https://www.pivotaltracker.com/story/show/166240987)

**Screenshots (if appropriate)**

N/A

**Questions:**

N/A

### CODE REVIEW

**Review by UhiriweAudace — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "express": "^4.16.3",
     "express-jwt": "^5.3.1",
     "express-session": "^1.15.6",
+    "global": "^4.3.2",
```
Comment (mifeille) [inline package.json]: I think that you added this package per accident. The right way of installing a package globally is ` yarn global add package-name `. Please keep in mind that you can simply use `yarn add ...` to install non-global packages.

Comment (EmyRukundo) [reply package.json]: alright, going to correct it

Comment (EmyRukundo) [reply package.json]: thanks, corrected it

**Review by deschantkn — APPROVED** [review]

**Review by Diama1 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -28,18 +28,19 @@
     "passport": "^0.4.0",
     "passport-local": "^1.0.0",
     "request": "^2.87.0",
+    "swagger-ui-express": "^4.0.6",
```
Comment (Quantum-35) [inline package.json]: Move `swagger-ui-express` to devDependencies

File: src/index.js [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
 import api from './api/routes';
 import globalMiddleware from './middleware/globalMiddleware';
+import  swaggerDoc from '../swagger.json';
+
 
 dotenv.config();
 const port = process.env.PORT || 3000;
 const app = express();
 
+app.use('/tesla-ah', swaggerUi.serve,swaggerUi.setup(swaggerDoc));
```
Comment (Quantum-35) [inline src/index.js]: Use the root URL `/` for the project documentation instead of `/tesla-ah`

**Review by Quantum-35 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (EmyRukundo) [reply package.json]: okay

File: src/index.js (hunk 3 above)
Comment (EmyRukundo) [reply src/index.js]: okay

**Review by eliemugenzi — COMMENTED** [review]
Check whether you have some merge conflicts with your package.json file

**Comment by EmyRukundo** [comment]
> Check whether you have some merge conflicts with your package.json file

resolved, thanks

**Review by eliemugenzi — APPROVED** [review]

**Review by Quantum-35 — APPROVED** [review]

**Pull request merged**